### ISSUE TITLE
Outputting class embeddings on tensor board

### ISSUE THREAD

**Issue by kielong** [mikeizbicki/cmc-csci181#52]
I'm a little confused about what command I'm supposed to run to output the class embeddings on tensorboard. I've already copied over the code for part 3, I'm just not sure what command I should run. I'm running "python3 names.py --warmstart='the model I trained' --embed". Is this the right command? Thanks!

**Comment by mikeizbicki** [comment]
Yes, that command should work.

**Comment by kielong** [comment]
I already ran the command. But now I'm on tensorboard, and there is no embeddings tab. Can I get some help with this?
![Screen Shot 2020-05-13 at 7 50 19 PM](https://user-images.githubusercontent.com/46983594/81880411-124bc080-9553-11ea-9588-01f3eea115e3.png)

**Comment by mikeizbicki** [comment]
`Projector` is the tab that you want.  I'm not sure why nothing would appear, though.  If there's no data, tensorboard will at least show a message saying so.

Do you get any error messages in the tensorboard terminal window?

**Comment by kielong** [comment]
No, I don't have any errors in the tensorboard terminal window.

**Comment by kielong** [comment]
Actually, I get the following:
W0513 21:01:53.525707 123145427017728 application.py:640] path /data/index.js not found, sending 404

**Comment by mikeizbicki** [comment]
`index.js` is part of the tensorboard package, and not part of your model output, so my best guess is that your installation is corrupted somehow.

**Comment by kielong** [comment]
I made a simple mistake. It works on Chrome, but not Safari. Should we submit the entire project folder or just the names.py file? Thanks!

**Comment by jordanhuard** [comment]
I am having a different but related problem. 

<img width="1258" alt="Screen Shot 2020-05-13 at 10 23 21 PM" src="https://user-images.githubusercontent.com/62961905/81896054-92c8ec00-9568-11ea-96f1-8facfe86da53.png">

TensorBoard keeps telling me that my points and dimension are loading. In the dropdown menu, there are 2 runs found. One is the one seen and the other is just completely blank. Switching between them changes nothing. 

In terminal, I am running:
```bash
python3 names.py --embed --warm_start=(log of my training) --log_dir=log
```
In my other tab where I call tensorboard, I encounter an error:
```bash
E0513 (time-stamp) (seq of numbers) plugin_event_multiplexer.py:226] Unable to reload accumulator '.': [Errno 2] No such file or directory: b'log/events.out.tfevents.1589423309.Jordans-MacBook-Pro-7.local.19474.0'
```
One of the files in the log folder is similar to this, but the numbers are different. There's a chance that tensorboard is looking for a file from an earlier run of the program, but I'm not sure how to fix this. I'm currently trying to load in this similar file through tensorboard on the browser. It currently says that it is parsing metadata, but it has been at this for about 10 minutes so I'm not sure if this is how to fix it. I will edit my comment if this ends up working.

My python script is directly from the video with a couple of changes I had to make to match up with my code.

**Comment by mikeizbicki** [comment]
@jordanhuard I think there is maybe some confusion about what `--log_dir` is doing.  

All of the output of your embedding code will be stored into whatever you pass in to `--log_dir`.  But this should NOT be the `log` folder.  

Tensorboard expects all runs to be in the form `part1/part2`.  When you run tensorboard, you specify the `part1` on the command line with the `--logdir` flag, and then each `part2` folder is listed as a one of the runs that gets displayed.  

You are setting `--log_dir=log`, so you are providing `part1`, but not `part2`.  Therefore tensorboard is getting confused.

Likely at this point, your `log` folder has lots of extraneous files inside of it, and so you'll have to clean it out in order to get tensorboard to work correctly.

**Comment by jordanhuard** [comment]
Ah I understand now. I made an ```embeddings``` folder and called ```log_dir=embeddings``` which fixed the problem. Thank you!
